src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterOptions.cs

```diff
@@ -43,6 +45,46 @@ public bool Indented
             }
         }
 
+        /// <summary>
+        /// Defines the indentation character used by <see cref="Utf8JsonWriter"/> when <see cref="Indented"/> is enabled. Defaults to the space character.
+        /// </summary>
+        /// <remarks>Allowed characters are space and horizontal tab.</remarks>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> contains an invalid character.</exception>
+        public char IndentCharacter
+        {
+            readonly get => (char)IndentByte;
+            set
+            {
+                JsonWriterHelper.ValidateIndentCharacter(value);
+                if (value is not JsonConstants.DefaultIndentCharacter)
+                    _optionsMask |= IndentCharacterBit;
+                else
+                    _optionsMask &= ~IndentCharacterBit;
+            }
+        }
+
+        /// <summary>
+        /// Defines the indentation size used by <see cref="Utf8JsonWriter"/> when <see cref="Indented"/> is enabled. Defaults to two.
+        /// </summary>
+        /// <remarks>Allowed values are integers between 0 and 127.</remarks>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="value"/> is out of the allowed range.</exception>
+        public int IndentSize
+        {
+            readonly get => HandleDefaultIndentSize((_optionsMask & IndentSizeMask) >> 3);
+            set
+            {
+                JsonWriterHelper.ValidateIndentSize(value);
+                _optionsMask = (_optionsMask & ~IndentSizeMask) + (HandleDefaultIndentSize(value) << 3);
+            }
+        }
+
+        private static int HandleDefaultIndentSize(int value) => value switch
```

Comment:
Nit: naming
```suggestion
        private static int EncodeIndentSize(int value) => value switch
```
Perhaps add a comment explaining that this is swapping `0` with the default value to ensure `default(JsonWriterOptions)` instances are well-defined. And that this implies that the encoder is also the decoder.

Reply:
Agreed.